### PR TITLE
A more complete Firerunner, with unbounded requests and responses to the VM

### PR DESCRIPTION
**Pull request by alevy**
This PR adds Vsock support to Firerunner, better structures the Firerunner code to pave the way towards a full controller, and completes the image creation tool to actually use the vsock to receive and respond to requests.

You can now use it as follows:

```bash
$ cd images/
$ ./mk_rtimage.sh nodejs nodejs.ext4
...
$ cd examples/loremjs
$ make
...
$ cd ../../firerunner
$ cat > test.jsonl # we'll add some dummy requests
{"foo": 1}
{"foo": 2}
{"foo": 3}
{"foo": 4}
$ cargo run --release --bin firerunner -- --rootfs ../images/nodejs.ext4 --appfs ../images/examples/loremjs/output.ext4 --kernel hello-vmlinux.bin < test.jsonl
Connection from VsockAddr { port: 1024, cid: 42 }
{"request":{"foo":1},"sentence":"Occaecat labore laboris officia laboris."}
{"request":{"foo":2},"sentence":"Minim ad pariatur id dolor consectetur non sunt nulla aute nisi elit dolore aute deserunt."}                                                                                
{"request":{"foo":3},"sentence":"Adipisicing ullamco Lorem enim ut pariatur cupidatat in cupidatat qui ullamco non."}                                                                                        
{"request":{"foo":4},"sentence":"Sint culpa sint aute et excepteur."}
```

### CODE REVIEW

**Review by tan-yue — APPROVED** [review]

**Comment by tan-yue** [comment]
Before this PR, I ran into the issue that write to the magic port caused segmentation fault when I used root file system generated by `mk_rtimage.sh`. Have you @alevy ever tested this?

**Comment by alevy** [comment]
Ah, I have not actually tried write to the io port with read only file systems. We can keep the devices read/write and resolve it separately

File: images/postscript.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ## Make sure fstab exists and is empty
-cat > /etc/fstab
+echo -n > /etc/fstab
```
Comment (LedgeDash) [inline images/postscript.sh:2]: sorry this is late, but what's fstab used for? is it related to vsock?

Comment (alevy) [reply images/postscript.sh:2]: This just overwrites fstab with an empty one so that Linux doesn’t mount any drives at boot.

**Pull request merged**